charts/streams-app/templates/_helpers.tpl

```diff
@@ -27,3 +27,20 @@ Create chart name and version as used by the chart label.
 {{- define "streams-app.chart" -}}
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{/*
+Helper function to add annotations to resources
+*/}}
+{{- define "streams-app.annotations" -}}
+{{- if or .Values.kafka.applicationId .Values.annotations }}
+  annotations:
+  {{- range $key, $value := .Values.annotations }}
+    {{ $key | quote }}: {{ $value | quote }}
+  {{- end }}
+
+  {{- if and .Values.kafka.applicationId (not .Values.annotations.consumerGroup) }}
+    consumerGroup: {{ .Values.kafka.applicationId | quote }}
+  {{- end }}
+{{- end }}
+{{- end }}
+
```

Comment:
I think it could be a little weird to have the consumerGroup annotation on every resource (like Service and ScaledObject). Maybe we can instead split this into two functions, one that prints all annotations in `.Values.annotations` and one that uses it to build the whole block. Then, most resources can stay like you've implemented it and only the deployment would need to have a custom implementation.
Just as pseudocode to maybe illustrate it a bit:
```tpl

{{- define "streams-app.default-annotations" -}}
  {{- range $key, $value := .Values.annotations }}
    {{ $key | quote }}: {{ $value | quote }}
  {{- end }}
{{- end }}

{{- define "streams-app.annotations" -}}
{{- if  .Values.annotations }}
  annotations:
   {{- include "streams-app.annotations" . | nindent 4 }}
{{- end }}
{{- end }}
```

Reply:
Good point! I have changed it. Here is the output of the deployment.yaml and service.yaml after template:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: bakdata-streams-app
  annotations:
    "MY_ANNOTATION": "foo-bar"
    "MY_ANNOTATION_2": "foo-bar-2"
    consumerGroup: "foo"
```

```yaml
apiVersion: v1
kind: Service
metadata:
  name: bakdata-streams-app
  annotations:
    "MY_ANNOTATION": "foo-bar"
    "MY_ANNOTATION_2": "foo-bar-2"
  labels:
    app: bakdata-streams-app
    chart: streams-app-0.1.0
    release: release-name
spec:
  selector:
    app: bakdata-streams-app
    release: release-name
  ports:
  type: ClusterIP
```